src/pybind/mgr/cephadm/module.py

```diff
@@ -646,13 +673,20 @@ def serve(self):
         # type: () -> None
         self.log.info("serve starting")
         while self.run:
+            self._check_hosts()
+
+            # refresh services
+            self.log.debug('refreshing services')
+            completion = self._get_services(maybe_refresh=True)
+            self._orchestrator_wait([completion])
```

Comment:
#32893

Reply:
Yeah... I tried changing this based on your suggested change to the upgrade PR a few weeks back but I couldn't make it work, and you were out on PTO or something, so I went with this for now.  Can you try fixing it in a separate pR?